sycl/doc/GetStartedWithSYCLCompiler.md

```diff
@@ -16,55 +16,102 @@ to install.
 SYCL support: follow the instructions under
 [SYCL* Compiler and Runtimes](https://github.com/intel/llvm/releases/tag/2019-07)
 
-# Build the SYCL compiler and runtime
+### Get the required tools:
+
+   a. `git` - for downloading the sources (Get it at https://git-scm.com/downloads)
+   
+   b. `cmake` - for building the compiler and tools, version 3.2 or later (Get it at: http://www.cmake.org/download)
+      
+   c. `python` - for building the compiler and running tests (Get it at: https://www.python.org/downloads/release/python-2716/ )
+
+   d. `Visual Studio 2017 or later` (Windows only. Get it at: https://visualstudio.microsoft.com/downloads/)
 
-Download the LLVM* repository with SYCL support to your local machine folder
-e.g. `$HOME/sycl` (assuming environment var `$SYCL_HOME`) folder using
-following command:
+
+# Configure environment:
+For simplicity it is assumed below that the environment variable SYCL_HOME contains path to a folder where the SYCL compiler and runtime will be stored.
+### Linux:
+```bash
+export SYCL_HOME=/export/home/workspaces/sycl_workspace
+```
+### Windows:
+Open a developer command prompt using one of tho methods:
+- Click start menu and search for the command prompt. So, for MSVC-2017 it is '`x64 Native Tools Command Prompt for VS 2017`'
+- run 'cmd' and then '`"C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Auxiliary\Build\vcvarsall.bat" x64`'
 
 ```bash
-git clone https://github.com/intel/llvm -b sycl $HOME/sycl
+set SYCL_HOME=%USERPROFILE%\workspaces\sycl_workspace
 ```
 
-Build the SYCL compiler and runtime following instruction below:
+# Checkout and Build the SYCL compiler and runtime
 
+### Linux:
 ```bash
+mkdir $SYCL_HOME
+cd $SYCL_HOME
+git clone https://github.com/intel/llvm -b sycl
 mkdir $SYCL_HOME/build
 cd $SYCL_HOME/build
+
 cmake -DCMAKE_BUILD_TYPE=Release \
 -DLLVM_EXTERNAL_PROJECTS="llvm-spirv;sycl" \
--DLLVM_EXTERNAL_SYCL_SOURCE_DIR=$SYCL_HOME/sycl \
--DLLVM_EXTERNAL_LLVM_SPIRV_SOURCE_DIR=$SYCL_HOME/llvm-spirv \
 -DLLVM_ENABLE_PROJECTS="clang;llvm-spirv;sycl" \
-$SYCL_HOME/llvm
+-DLLVM_EXTERNAL_SYCL_SOURCE_DIR=$SYCL_HOME/llvm/sycl \
+-DLLVM_EXTERNAL_LLVM_SPIRV_SOURCE_DIR=$SYCL_HOME/llvm/llvm-spirv \
+$SYCL_HOME/llvm/llvm
+
 make -j`nproc` sycl-toolchain
 ```
+### Windows:
+```bash
+mkdir %SYCL_HOME%
+cd %SYCL_HOME%
+git clone https://github.com/intel/llvm -b sycl
+mkdir %SYCL_HOME%\build
+cd %SYCL_HOME%\build
+
+cmake -G "Ninja" -DCMAKE_BUILD_TYPE=Release -DLLVM_EXTERNAL_PROJECTS="llvm-spirv;sycl" -DLLVM_ENABLE_PROJECTS="clang;llvm-spirv;sycl" -DLLVM_EXTERNAL_SYCL_SOURCE_DIR="%SYCL_HOME%\llvm\sycl" -DLLVM_EXTERNAL_LLVM_SPIRV_SOURCE_DIR="%SYCL_HOME%\llvm\llvm-spirv" -DLLVM_TARGETS_TO_BUILD="X86" -DCMAKE_C_COMPILER=cl -DCMAKE_CXX_COMPILER=cl -DCMAKE_C_FLAGS="/GS" -DCMAKE_CXX_FLAGS="/GS" -DCMAKE_EXE_LINKER_FLAGS="/NXCompat /DynamicBase" -DCMAKE_SHARED_LINKER_FLAGS="/NXCompat /DynamicBase" "%SYCL_HOME%\llvm\llvm"
```

Comment:
1. Is there any reason why we suggest using ninja on windows but make on windows?
2. Why do you set -DLLVM_TARGETS_TO_BUILD="X86" only on windows?

Reply:
regarding 'ninja'. I answered that above. I had 3 reasons:
a) it is already included into MSVC-2017 and 2019. No need to download/install one more tool - 'make'
b) ninja is much faster for incremental/subsequent builds
c) ninja is recommended at http://clang.llvm.org/get_started.html

I'll check if -DLLVM_TARGETS_TO_BUILD="X86" can be removed from this 'cmake' command.

Reply:
1. I meant why don't we suggest using ninja on Linux as well?

Reply:
why not ninja on Linux..., well..., a) historically, we use it for so long on Linux and everything works great
b) make on Linux is recommended at http://clang.llvm.org/get_started.html (I don't know if there are any underwater rocks when using ninja on Linux)

Reply:
Regarding "-DLLVM_TARGETS_TO_BUILD="X86" on Windows:
If you do not set the target, then all targets are built. I suppose, it does not make sense building all of them (it is about +1000 additional sources files to compile).
On Linux where it is not set explicitly now I see:
09:11:22 > llvm-config --targets-built
AArch64 AMDGPU ARM BPF Hexagon Lanai Mips MSP430 NVPTX PowerPC RISCV Sparc SystemZ WebAssembly X86 XCore

Reply:
Probably we need update cmake line for linux to include this define.

Reply:
I agree; will add -DLLVM_TARGETS_TO_BUILD="X86" to Linux cmake comand.